src/libPMacc/include/pluginSystem/PluginConnector.hpp

```diff
@@ -180,6 +181,19 @@ namespace PMacc
             }
         }
 
+        template<typename Plugin>
+        Plugin* getPluginFromType()
```

Comment:
I checked if it is possible that more than one plugin can have the same type. It is possible e.g., in PIConGPU the plugin `Visualisation<ParticleType,OuputMethod>` is registered multiple times with the same type signature.
This means that the name must be `getFirstPluginFromType()` or it should be return a vector with all instances. The second version is IMO the cleanest solution.